ddtrace/profiling/collector/_traceback.pyx

```diff
@@ -1,61 +1,22 @@
-IF PY_MAJOR_VERSION > 3 or (PY_MAJOR_VERSION == 3 and PY_MINOR_VERSION >= 11):
-    from cpython.object cimport PyObject
-    from cpython.ref cimport Py_XDECREF
-
-    cdef extern from "<Python.h>":
-        ctypedef struct PyCodeObject:
-            pass
-        ctypedef struct PyFrameObject:
-            pass
-        PyObject* PyDict_GetItem(PyObject* p, PyObject* key)
-        PyObject* PyTuple_GetItem(PyObject* p, Py_ssize_t pos)
-        Py_ssize_t PyTuple_Size(PyObject* p)
-
-        PyCodeObject* PyFrame_GetCode(PyFrameObject* frame)
-        PyFrameObject* PyFrame_GetBack(PyFrameObject* frame)
-        PyObject* PyFrame_GetLocals(PyFrameObject* frame)
-        int PyFrame_GetLineNumber(PyFrameObject* frame)
-        PyObject* PyCode_GetVarnames(PyCodeObject* co)
-
-
-cpdef _extract_class_name(frame):
+cdef str _extract_class_name(object frame):
     # type: (...) -> str
     """Extract class name from a frame, if possible.
 
     :param frame: The frame object.
     """
-    # Python 3.11 moved PyFrameObject members to internal C API and cannot be directly accessed
-    IF PY_MAJOR_VERSION > 3 or (PY_MAJOR_VERSION == 3 and PY_MINOR_VERSION >= 11):
-        code = PyFrame_GetCode(<PyFrameObject*>frame)  # Returns strong reference
-        co_varnames = PyCode_GetVarnames(code)  # Returns new reference
-        if co_varnames == NULL or PyTuple_Size(co_varnames) == 0:
-            Py_XDECREF(<PyObject*>code)
-            return ""
-        argname = PyTuple_GetItem(co_varnames, 0)
-        Py_XDECREF(<PyObject*>code)
-        Py_XDECREF(co_varnames)
+    if frame.f_code.co_varnames:
+        argname = frame.f_code.co_varnames[0]
```

Comment:
are we sure we'll have at least 1?

Reply:
then `frame.f_code.co_varnames` would evaluate to `False`, no? This is the current coding btw

Reply:
this was pre-coffee 🤦🏻